src/credentials.rs

```diff
@@ -2,21 +2,39 @@
 
 use eyre::{Context, ContextCompat};
 use serde::{Deserialize, Serialize};
-use std::{env, path::PathBuf};
+use std::{
+    collections::HashMap,
+    env,
+    ops::{Deref, DerefMut},
+    path::PathBuf,
+};
 use tokio::fs;
 
-use crate::registry::ArtifactoryConfig;
+use crate::manifest::{RegistryToken, RegistryUri};
 
 /// Global configuration directory for `buffrs`
 pub const BUFFRS_HOME: &str = ".buffrs";
 /// Filename of the credential store
 pub const CREDENTIALS_FILE: &str = "credentials.toml";
 
 /// Credential store for storing authentication data
-#[derive(Debug, Default, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[derive(Debug, Default, Clone)]
 pub struct Credentials {
-    /// Artifactory credentials
-    pub artifactory: Option<ArtifactoryConfig>,
+    credentials: HashMap<RegistryUri, RegistryToken>,
+}
+
+impl Deref for Credentials {
```

Comment:
Overusing deref / implementing inhertiance through deref is an antipattern – see https://rust-unofficial.github.io/patterns/anti_patterns/deref.html